pynemo/tide/nemo_bdy_tide3.py

```diff
@@ -1,272 +1,291 @@
+    #extract the surface elevation at each z-point
+    tpxo_z = tpxo_extract_HC.TpxoExtract(setup.settings, dst_lat, dst_lon, g_type)
+    #convert back the z-longitudes into the usual conventions (-180E/+180E)
+    dst_lon[dst_lon > 180.0] = dst_lon[dst_lon > 180.0]-360.0
+    #check if elevation data are missing
+    ind = np.where((np.isnan(tpxo_z.amp)) | (np.isnan(tpxo_z.gph)))
+    if ind[0].size > 0:
+        logger.warning('Missing elveation along the open boundary')
+
+    ampz = tpxo_z.amp
+    phaz = tpxo_z.gph
+    ampz[ind] = 0.0
+    phaz[ind] = 0.0
+
+    #extract U values of constituents
+    dst_lon = DC.bdy_lonlat[Grid_U.grid_type]['lon'][Grid_U.bdy_r == 0]
+    dst_lat = DC.bdy_lonlat[Grid_U.grid_type]['lat'][Grid_U.bdy_r == 0]
+    #set the array size for the target boundary output
+    if len(dst_lon) != len(dst_lon): logger.error('These should be the same size')
+    else: nbdyu = len(dst_lon)
+
+    #convert the U-longitudes into the TMD conventions (0/360E)
+    dst_lon[dst_lon < 0.0] = dst_lon[dst_lon < 0.0]+360.0
+
+    tpxo_ux = tpxo_extract_HC.TpxoExtract(setup.settings, dst_lat, dst_lon, Grid_U.grid_type)
+    tpxo_vx = tpxo_extract_HC.TpxoExtract(setup.settings, dst_lat, dst_lon, Grid_V.grid_type)
+
+    ampuX = tpxo_ux.amp
+    phauX = tpxo_ux.gph
+    ampvX = tpxo_vx.amp
+    phavX = tpxo_vx.gph
+
+    #check if ux data are missing
+    ind = np.where((np.isnan(ampuX)) | (np.isnan(phauX)))
+    if ind[0].size > 0:
+        logger.warning('Missing zonal velocity along the x open boundary')
+    ampuX[ind] = 0
+    phauX[ind] = 0
+    #check if vx data are missing
+    ind = np.where((np.isnan(ampvX)) | (np.isnan(phavX)))
+    if ind[0].size > 0:
+        logger.warning('Missing meridional velocity along the x open boundary')
+    ampvX[ind] = 0
+    phavX[ind] = 0
+
+    #convert back the u-longitudes into the usual conventions (-180E/+180E)
+    dst_lon[dst_lon > 180.0] = dst_lon[dst_lon > 180.0]-360.0
+
+    #extract V values of constituents
+    dst_lon = DC.bdy_lonlat[Grid_V.grid_type]['lon'][Grid_V.bdy_r == 0]
+    dst_lat = DC.bdy_lonlat[Grid_V.grid_type]['lat'][Grid_V.bdy_r == 0]
+    #set the array size for the target boundary output
+    if len(dst_lon) != len(dst_lon): logger.error('These should be the same size')
+    else: nbdyv = len(dst_lon)
+
+    #convert the U-longitudes into the TMD conventions (0/360E)
+    dst_lon[dst_lon < 0.0] = dst_lon[dst_lon < 0.0]+360.0
+    tpxo_uy = tpxo_extract_HC.TpxoExtract(setup.settings, dst_lat, dst_lon, Grid_U.grid_type)
+    tpxo_vy = tpxo_extract_HC.TpxoExtract(setup.settings, dst_lat, dst_lon, Grid_V.grid_type)
+
+    ampuY = tpxo_uy.amp
+    phauY = tpxo_uy.gph
+    ampvY = tpxo_vy.amp
+    phavY = tpxo_vy.gph
+
+    #check if uy data are missing
+    ind = np.where((np.isnan(ampuY)) | (np.isnan(phauY)))
+    if ind[0].size > 0:
+        logger.warning('Missing zonal velocity along the y open boundary')
```

Comment:
@jdha I've updated this warning message can you confirm this is the correct interpretation?